### ISSUE TITLE
%% string breaks printf

### ISSUE THREAD

**Issue by linehill** [intel/compute-runtime#536]
Link to OpenCL reproducer: https://github.com/pocl/pocl/blob/master/tests/spirv/printf-kernels.cl#L55.

Instead of printing the expected output: 
```
hello
%
hello % world
%s
```
It print something like this instead (output may vary):
```
hello
%%\n,%\n,%\n,%\n,%\n,%\n,%\n,%\n,%\n,%\n,%\n
```
Test environment:
OS: Ubuntu 20.04.4 LTS
Intel Compute Runtime version: 22.16.22992
Device: Intel(R) HD Graphics 530 [0x1912]

**Comment by JablonskiMateusz** [comment]
Hi @linehill 
It should be fixed within https://github.com/intel/compute-runtime/commit/20bb97ece3feb40270bd787c752a200ba9092db8 
Could you confirm that it works fine when using this change or recent release?

**Comment by linehill** [comment]
Confirmed to be fixed in release 22.42.24548.
